### ISSUE TITLE
WARNING: Attempt to (de-)serialize anonymous class hudson.plugins.selenium.PluginImpl$1

### ISSUE THREAD

**Issue by staniere** [jenkinsci/selenium-plugin#130]
Hello, 
since friday 6 july, I encountered some problems with my selenium grid. My grid did not accept new connections to chrome browser. 

I think this warning message may be a cause to this problem (warning displayed in "selenium hub console" just after clicking on Restart) : ("channel stopped" displayed without any session request)
```
Starting Selenium Grid
[Jenkins] $ "C:\Program Files (x86)\Jenkins\jre\bin\java" -cp "C:\Program Files (x86)\Jenkins\war\WEB-INF\lib\remoting-3.20.jar" hudson.remoting.Launcher -cp D:\Integration\Jenkins\plugins\selenium\WEB-INF\lib\selenium-server-standalone-3.12.0.jar;D:\Integration\Jenkins\plugins\selenium\WEB-INF\lib\htmlunit-driver-standalone-2.29.2.jar -connectTo localhost:52996
Starting Selenium Grid
channel started
juil. 09, 2018 10:21:06 AM hudson.plugins.selenium.HubLauncher call
OFF: Grid hub starting with log level INFO
juil. 09, 2018 10:21:06 AM hudson.plugins.selenium.HubLauncher call
OFF: Grid Hub preparing to start on port 4444
2018-07-09 10:21:07.255:INFO::pool-1-thread-1 for channel id=152: Logging initialized @3135ms to org.seleniumhq.jetty9.util.log.StdErrLog
2018-07-09 10:21:07.349:INFO:osjs.Server:pool-1-thread-1 for channel id=152: jetty-9.4.7.v20170914, build timestamp: 2017-11-21T22:27:37+01:00, git hash: 82b8fb23f757335bb3329d540ce37a2a2615f0a8
2018-07-09 10:21:07.380:INFO:osjs.session:pool-1-thread-1 for channel id=152: DefaultSessionIdManager workerName=node0
2018-07-09 10:21:07.380:INFO:osjs.session:pool-1-thread-1 for channel id=152: No SessionScavenger set, using defaults
2018-07-09 10:21:07.380:INFO:osjs.session:pool-1-thread-1 for channel id=152: Scavenging every 660000ms
2018-07-09 10:21:07.396:INFO:osjsh.ContextHandler:pool-1-thread-1 for channel id=152: Started o.s.j.s.ServletContextHandler@524525{/,null,AVAILABLE}
2018-07-09 10:21:07.411:INFO:osjs.AbstractConnector:pool-1-thread-1 for channel id=152: Started ServerConnector@1a7acdc{HTTP/1.1,[http/1.1]}{sgemajo2:4444}
2018-07-09 10:21:07.411:INFO:osjs.Server:pool-1-thread-1 for channel id=152: Started @3288ms
juil. 09, 2018 10:21:07 AM org.openqa.grid.web.Hub start
INFO: Selenium Grid hub is up and running
juil. 09, 2018 10:21:07 AM org.openqa.grid.web.Hub start
INFO: Nodes should register to http://sgemajo2:4444/grid/register/
juil. 09, 2018 10:21:07 AM org.openqa.grid.web.Hub start
INFO: Clients should connect to http://sgemajo2:4444/wd/hub
juil. 09, 2018 10:21:07 AM hudson.plugins.selenium.HubLauncher call
OFF: Grid Hub started on port 4444 with args: -maxSession 5 -timeout 300 -browserTimeout 0 -host sgemajo2
juil. 09, 2018 10:21:07 AM org.jenkinsci.remoting.util.AnonymousClassWarnings warn
WARNING: Attempt to (de-)serialize anonymous class hudson.plugins.selenium.PluginImpl$1; see: https://jenkins.io/redirect/serialization-of-anonymous-classes/
channel stopped
```

here is another "selenium hub console" log when restarting selenium hub and trying to connect to it with selenium : ("channel stopped" displayed after new session request)

```
Starting Selenium Grid
[Jenkins] $ "C:\Program Files (x86)\Jenkins\jre\bin\java" -cp "C:\Program Files (x86)\Jenkins\war\WEB-INF\lib\remoting-3.20.jar" hudson.remoting.Launcher -cp D:\Integration\Jenkins\plugins\selenium\WEB-INF\lib\selenium-server-standalone-3.12.0.jar;D:\Integration\Jenkins\plugins\selenium\WEB-INF\lib\htmlunit-driver-standalone-2.29.2.jar -connectTo localhost:56257
Starting Selenium Grid
channel started
juil. 09, 2018 9:21:21 AM hudson.plugins.selenium.HubLauncher call
OFF: Grid hub starting with log level INFO
juil. 09, 2018 9:21:21 AM hudson.plugins.selenium.HubLauncher call
OFF: Grid Hub preparing to start on port 4444
2018-07-09 09:21:22.407:INFO::pool-1-thread-3 for channel id=348: Logging initialized @3142ms to org.seleniumhq.jetty9.util.log.StdErrLog
2018-07-09 09:21:22.501:INFO:osjs.Server:pool-1-thread-3 for channel id=348: jetty-9.4.7.v20170914, build timestamp: 2017-11-21T22:27:37+01:00, git hash: 82b8fb23f757335bb3329d540ce37a2a2615f0a8
2018-07-09 09:21:22.532:INFO:osjs.session:pool-1-thread-3 for channel id=348: DefaultSessionIdManager workerName=node0
2018-07-09 09:21:22.532:INFO:osjs.session:pool-1-thread-3 for channel id=348: No SessionScavenger set, using defaults
2018-07-09 09:21:22.532:INFO:osjs.session:pool-1-thread-3 for channel id=348: Scavenging every 660000ms
2018-07-09 09:21:22.532:INFO:osjsh.ContextHandler:pool-1-thread-3 for channel id=348: Started o.s.j.s.ServletContextHandler@17f3b63{/,null,AVAILABLE}
2018-07-09 09:21:22.564:INFO:osjs.AbstractConnector:pool-1-thread-3 for channel id=348: Started ServerConnector@1453616{HTTP/1.1,[http/1.1]}{sgemajo2:4444}
2018-07-09 09:21:22.564:INFO:osjs.Server:pool-1-thread-3 for channel id=348: Started @3290ms
juil. 09, 2018 9:21:22 AM org.openqa.grid.web.Hub start
INFO: Selenium Grid hub is up and running
juil. 09, 2018 9:21:22 AM org.openqa.grid.web.Hub start
INFO: Nodes should register to http://sgemajo2:4444/grid/register/
juil. 09, 2018 9:21:22 AM org.openqa.grid.web.Hub start
INFO: Clients should connect to http://sgemajo2:4444/wd/hub
juil. 09, 2018 9:21:22 AM hudson.plugins.selenium.HubLauncher call
OFF: Grid Hub started on port 4444 with args: -maxSession 5 -timeout 300 -browserTimeout 0 -host sgemajo2
juil. 09, 2018 9:21:22 AM org.jenkinsci.remoting.util.AnonymousClassWarnings warn
WARNING: Attempt to (de-)serialize anonymous class hudson.plugins.selenium.PluginImpl$1; see: https://jenkins.io/redirect/serialization-of-anonymous-classes/
juil. 09, 2018 9:29:10 AM org.openqa.grid.web.servlet.handler.RequestHandler process
INFO: Got a request to create a new session: Capabilities {browserName: chrome}
juil. 09, 2018 9:29:10 AM org.openqa.grid.web.servlet.handler.RequestHandler process
INFO: Error forwarding the new session Empty pool of VM for setup Capabilities {browserName: chrome}
org.openqa.grid.common.exception.GridException: Empty pool of VM for setup Capabilities {browserName: chrome}
	at org.openqa.grid.internal.ProxySet.verifyAbilityToHandleDesiredCapabilities(ProxySet.java:146)
	at org.openqa.grid.internal.DefaultGridRegistry.addNewSessionRequest(DefaultGridRegistry.java:217)
	at org.openqa.grid.web.servlet.handler.RequestHandler.process(RequestHandler.java:111)
	at org.openqa.grid.web.servlet.DriverServlet.process(DriverServlet.java:86)
	at org.openqa.grid.web.servlet.DriverServlet.doPost(DriverServlet.java:70)
	at javax.servlet.http.HttpServlet.service(HttpServlet.java:707)
	at javax.servlet.http.HttpServlet.service(HttpServlet.java:790)
	at org.seleniumhq.jetty9.servlet.ServletHolder.handle(ServletHolder.java:860)
	at org.seleniumhq.jetty9.servlet.ServletHandler.doHandle(ServletHandler.java:535)
	at org.seleniumhq.jetty9.server.handler.ScopedHandler.handle(ScopedHandler.java:143)
	at org.seleniumhq.jetty9.security.SecurityHandler.handle(SecurityHandler.java:548)
	at org.seleniumhq.jetty9.server.handler.HandlerWrapper.handle(HandlerWrapper.java:132)
	at org.seleniumhq.jetty9.server.handler.ScopedHandler.nextHandle(ScopedHandler.java:190)
	at org.seleniumhq.jetty9.server.session.SessionHandler.doHandle(SessionHandler.java:1595)
	at org.seleniumhq.jetty9.server.handler.ScopedHandler.nextHandle(ScopedHandler.java:188)
	at org.seleniumhq.jetty9.server.handler.ContextHandler.doHandle(ContextHandler.java:1253)
	at org.seleniumhq.jetty9.server.handler.ScopedHandler.nextScope(ScopedHandler.java:168)
	at org.seleniumhq.jetty9.servlet.ServletHandler.doScope(ServletHandler.java:473)
	at org.seleniumhq.jetty9.server.session.SessionHandler.doScope(SessionHandler.java:1564)
	at org.seleniumhq.jetty9.server.handler.ScopedHandler.nextScope(ScopedHandler.java:166)
	at org.seleniumhq.jetty9.server.handler.ContextHandler.doScope(ContextHandler.java:1155)
	at org.seleniumhq.jetty9.server.handler.ScopedHandler.handle(ScopedHandler.java:141)
	at org.seleniumhq.jetty9.server.handler.HandlerWrapper.handle(HandlerWrapper.java:132)
	at org.seleniumhq.jetty9.server.Server.handle(Server.java:530)
	at org.seleniumhq.jetty9.server.HttpChannel.handle(HttpChannel.java:347)
	at org.seleniumhq.jetty9.server.HttpConnection.onFillable(HttpConnection.java:256)
	at org.seleniumhq.jetty9.io.AbstractConnection$ReadCallback.succeeded(AbstractConnection.java:279)
	at org.seleniumhq.jetty9.io.FillInterest.fillable(FillInterest.java:102)
	at org.seleniumhq.jetty9.io.ChannelEndPoint$2.run(ChannelEndPoint.java:124)
	at org.seleniumhq.jetty9.util.thread.strategy.EatWhatYouKill.doProduce(EatWhatYouKill.java:247)
	at org.seleniumhq.jetty9.util.thread.strategy.EatWhatYouKill.produce(EatWhatYouKill.java:140)
	at org.seleniumhq.jetty9.util.thread.strategy.EatWhatYouKill.run(EatWhatYouKill.java:131)
	at org.seleniumhq.jetty9.util.thread.ReservedThreadExecutor$ReservedThread.run(ReservedThreadExecutor.java:382)
	at org.seleniumhq.jetty9.util.thread.QueuedThreadPool.runJob(QueuedThreadPool.java:708)
	at org.seleniumhq.jetty9.util.thread.QueuedThreadPool$2.run(QueuedThreadPool.java:626)
	at java.lang.Thread.run(Unknown Source)

channel stopped
```

Here is log of selenium test when trying to connect to jenkins selenium grid : 
```
Connexion au hub Grids sur http://sgemajo2:4444/wd/hub
[ERROR] Tests run: 2, Failures: 0, Errors: 2, Skipped: 0, Time elapsed: 1.004 s <<< FAILURE! - in tests.TestClaims
[ERROR] Cla1AddClaimOn1Order(tests.TestClaims)  Time elapsed: 1.004 s  <<< ERROR!
org.openqa.selenium.WebDriverException: 
Error forwarding the new session Empty pool of VM for setup Capabilities {browserName: chrome}
Command duration or timeout: 339 milliseconds
Caused by: org.openqa.selenium.WebDriverException: 
Error forwarding the new session Empty pool of VM for setup Capabilities {browserName: chrome}
Build info: version: '3.12.0', revision: '7c6e0b3', time: '2018-05-08T14:04:26.12Z'
System info: host: 'sgeqmfee1', ip: '10.0.9.8', os.name: 'Linux', os.arch: 'amd64', os.version: '2.6.32-573.7.1.el6.x86_64', java.version: '1.8.0_161'
Driver info: driver.version: unknown
```

**Comment by Mobrockers** [comment]
This is not related to that warning

**Comment by miguelconde91** [comment]
I have the same issue:

`Starting Selenium Grid
[jenkins] $ /usr/lib/jvm/java-8-openjdk-amd64/jre/bin/java -cp /var/cache/jenkins/war/WEB-INF/lib/remoting-3.21.1.jar hudson.remoting.Launcher -cp /var/lib/jenkins/plugins/selenium/WEB-INF/lib/selenium-server-standalone-3.12.0.jar:/var/lib/jenkins/plugins/selenium/WEB-INF/lib/htmlunit-driver-standalone-2.29.2.jar -connectTo localhost:32881
Starting Selenium Grid
channel started
Nov 02, 2018 3:58:37 PM hudson.plugins.selenium.HubLauncher call
OFF: Grid hub starting with log level INFO
Nov 02, 2018 3:58:37 PM hudson.plugins.selenium.HubLauncher call
OFF: Grid Hub preparing to start on port 4444
2018-11-02 15:58:38.219:INFO::pool-1-thread-1 for channel id=26261: Logging initialized @2408ms to org.seleniumhq.jetty9.util.log.StdErrLog
2018-11-02 15:58:38.286:INFO:osjs.Server:pool-1-thread-1 for channel id=26261: jetty-9.4.7.v20170914, build timestamp: 2017-11-21T21:27:37Z, git hash: 82b8fb23f757335bb3329d540ce37a2a2615f0a8
2018-11-02 15:58:38.317:INFO:osjs.session:pool-1-thread-1 for channel id=26261: DefaultSessionIdManager workerName=node0
2018-11-02 15:58:38.317:INFO:osjs.session:pool-1-thread-1 for channel id=26261: No SessionScavenger set, using defaults
2018-11-02 15:58:38.319:INFO:osjs.session:pool-1-thread-1 for channel id=26261: Scavenging every 660000ms
2018-11-02 15:58:38.325:INFO:osjsh.ContextHandler:pool-1-thread-1 for channel id=26261: Started o.s.j.s.ServletContextHandler@3cc6e6da{/,null,AVAILABLE}
Nov 02, 2018 3:58:38 PM org.openqa.grid.internal.DefaultGridRegistry assignRequestToProxy
INFO: Shutting down registry.
2018-11-02 15:58:38.403:INFO:osjs.AbstractConnector:pool-1-thread-1 for channel id=26261: Stopped ServerConnector@71fb4ca8{HTTP/1.1,[http/1.1]}{jenkins.mydomain.com:4444}
2018-11-02 15:58:38.403:INFO:osjs.session:pool-1-thread-1 for channel id=26261: Stopped scavenging
2018-11-02 15:58:38.404:INFO:osjsh.ContextHandler:pool-1-thread-1 for channel id=26261: Stopped o.s.j.s.ServletContextHandler@3cc6e6da{/,null,UNAVAILABLE}
Nov 02, 2018 3:58:38 PM org.openqa.grid.web.Hub start
SEVERE: Port 4444 is busy, please choose a free port for the hub and specify it using -port option
Nov 02, 2018 3:58:38 PM hudson.plugins.selenium.HubLauncher call
OFF: Grid Hub started on port 4444 with args: -maxSession 5 -timeout 300 -browserTimeout 0 -host jenkins.mydomain.com
Nov 02, 2018 3:58:39 PM org.jenkinsci.remoting.util.AnonymousClassWarnings warn
WARNING: Attempt to (de-)serialize anonymous class hudson.plugins.selenium.PluginImpl$1; see: https://jenkins.io/redirect/serialization-of-anonymous-classes/
channel stopped`

**Comment by staniere** [comment]
Hello, 
sorry to let this ticket open. 
as Mobrockers said, this problem was not due to this warning "WARNING: Attempt to (de-)serialize anonymous class hudson.plugins.selenium.PluginImpl$1; see: https://jenkins.io/redirect/serialization-of-anonymous-classes/"
If i remember well, it was linked to bad browser version versus selenium webdriver. 
Finally, i had to reboot webserver, and then jenkins grid plugin was ok again.

**Comment by staniere** [comment]
miguelconde91, I think your error is not the same than me. I suppose you should look at this error : 
"org.openqa.grid.web.Hub start SEVERE: Port 4444 is busy, please choose a free port for the hub and specify it using -port option" which seems to say that your port is already taken by another grid or program.

**Comment by miguelconde91** [comment]
However my port 4444 is not busy, with netstat -tnlp I can not see this port in use. Thanks by your response

**Comment by staniere** [comment]
when trying another port, you always have same error ?

**Comment by miguelconde91** [comment]
My problem is like this https://github.com/jenkinsci/selenium-plugin/issues/128 was fixed when I changed the hostname to 127.0.0.1, but this alternative make not visible the 4444 port to the slaves
